core/src/main/scala/akka/persistence/cassandra/journal/CassandraJournal.scala

```diff
@@ -428,15 +479,57 @@ private[cassandra] object CassandraJournal {
   private case class MessageId(persistenceId: String, sequenceNr: Long)
   private case class SerializedAtomicWrite(persistenceId: String, payload: Seq[Serialized])
   private case class Serialized(persistenceId: String, sequenceNr: Long, serialized: ByteBuffer, tags: Set[String],
-                                eventManifest: String, serManifest: String, serId: Int, writerUuid: String)
+                                eventAdapterManifest: String, serManifest: String, serId: Int, writerUuid: String,
+                                meta: Option[SerializedMeta])
+  private case class SerializedMeta(serialized: ByteBuffer, serManifest: String, serId: Int)
   private case class PartitionInfo(partitionNr: Long, minSequenceNr: Long, maxSequenceNr: Long)
 
-  def deserializeEvent(serialization: Serialization, row: Row): Any = {
-    serialization.deserialize(
-      row.getBytes("event").array,
-      row.getInt("ser_id"),
-      row.getString("ser_manifest")
-    ).get
+  class EventDeserializer(serialization: Serialization) {
+
+    // cache to avoid repeated check via ColumnDefinitions
+    @volatile private var _hasMetaColumns: Option[Boolean] = None
```

Comment:
since this is internal I think it's safe to assume that it's only used from places where this holds. E.g. there is only one table on the write side, and on the query side (the materialized view) a separate EventDeserializer instance is used.